ambry-api/src/test/java/com.github.ambry/router/GetBlobRangeTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.router;
+
+import junit.framework.Assert;
+import org.junit.Test;
+
+import static com.github.ambry.router.GetBlobRange.UNDEFINED_OFFSET;
+import static org.junit.Assert.*;
+
+
+public class GetBlobRangeTest {
+  @Test
+  public void testValidRange() {
+    // 0-0 (0th byte)
+    GetBlobRange range = new GetBlobRange(0, 0);
+    assertFalse(range.withinTotalSize(0));
+    assertFalse(range.withinTotalSize(-1));
+    assertTrue(range.withinTotalSize(2));
+    assertEquals(0, range.getStartOffset(2));
+    assertEquals(0, range.getEndOffset(2));
+
+    // 0- (bytes after/including 0)
+    range = new GetBlobRange(0, UNDEFINED_OFFSET);
+    assertFalse(range.withinTotalSize(0));
+    assertFalse(range.withinTotalSize(-1));
+    assertTrue(range.withinTotalSize(20));
+    assertEquals(0, range.getStartOffset(20));
+    assertEquals(19, range.getEndOffset(20));
+
+    // 15- (bytes after/including 15)
+    range = new GetBlobRange(15, UNDEFINED_OFFSET);
+    assertFalse(range.withinTotalSize(15));
+    assertFalse(range.withinTotalSize(-1));
+    assertTrue(range.withinTotalSize(20));
+    assertEquals(15, range.getStartOffset(20));
+    assertEquals(19, range.getEndOffset(20));
+    assertTrue(range.withinTotalSize(16));
+    assertEquals(15, range.getStartOffset(16));
+    assertEquals(15, range.getEndOffset(16));
+
+    // -20 (last 20 bytes)
```

Comment:
Right now, that fails in the constructor, so there's a test for it in `testInvalidRanges()`, but I'm looking into changing this behavior.